### ISSUE TITLE
WebVTT

### ISSUE THREAD

**Issue by jgraham** [web-platform-tests/interop#860]
### Description

WebVTT is a feature to display text track synchronized to videos. Very often those are subtitles, but don't have to be.

Many major websites don't use WebVTT and instead often use custom solutions, probably due to poor interoperability in this area. This means that it's harder to have features such as PiP work well with subtitles. Custom solutions also tend to sacrifice some  accessibility (WebVTT is declarative, and standardized, so tools can process it with ease). 

## Tests
https://wpt.fyi/results/webvtt?label=experimental&label=master&aligned

### Specification

https://w3c.github.io/webvtt/

**Comment by jernoble** [comment]
From the dup'd issue #874:

> ### Additional Signals
> 
> This proposal [was discussed](https://www.w3.org/2024/09/27-tt-minutes.html#t09) at the Timed Text WG at TPAC this year, and the group unanimously supported its inclusion in Interop 2025.
> 
> There is an existing [test suite](https://github.com/web-platform-tests/wpt/tree/master/webvtt/rendering/cues-with-video/processing-model) for WebVTT Rendering. [Low scores](https://wpt.fyi/results/webvtt/rendering?label=experimental&label=master&aligned) demonstrate a severe lack of interoperability between browsers, with no browser passing more than 30% of the tests.

**Comment by jernoble** [comment]
In the US, there is a regulatory requirement to allow users to modify the default styling of captions. I can only speak with experience about Apple platforms, but on macOS, this functionality is baked into the operating system itself. Both Chrome and Safari use platform APIs to read the users' preferences for caption rendering, which is where these styles in your example come from.

If you go into System Settings -> Accessibility -> Captions, and change the style from the default of "Transparent Background", (and optionally restart Chrome), you should see those changes reflected in both Chrome and Safari's default caption rendering.

So I would disagree with this statement:

> Chromium and WebKit's padding, rounded corners, non-black background color, and block box layout, are technically spec violations.

This is true only in the very narrowest sense. There are settings and preferences, many about accessibility, that change the default rendering of web pages.

**Comment by jernoble** [comment]
This also, IMO, misses the forest for the trees. The problem with WebVTT interop is not that the _default_ caption styles differ between UAs, but that captions with _explicit_ styles set by the page differ between UAs.

**Comment by zcorpan** [comment]
Yes, the "Classic" setting in macOS looks closer to what the spec says.

Is it acceptable to switch away from the user's preference when there is some author CSS applied (e.g. color)?

**Comment by jernoble** [comment]
> Is it acceptable to switch away from the user's preference when there is some author CSS applied (e.g. color)?

It seems like that's exactly what CSS is for. If the page wants a specific text color, background, font family, font size, etc., they can add styles to do so. 

(This answer is complicated somewhat, because the Accessibility Settings for macOS and iOS allow the user to say _"don't allow the site to change my font color"_ for example.  But the default styles allow everything to be overridden.)

For WPT/Interop, this issue could be resolved by applying explicit styles to the cues under test.

**Comment by btsimonh** [comment]
My huge bugbears about WebVTT are:
1/ You cannot make a single file to reproduce vertical position on different browsers.  i.e. Chrome's positioning is different to Firefox is different to Safari is different to Opera.
2/ Horizontal positioning is equally compromised and different.
3/ Although the spec indicates you could anchor against a bottom position or a top position (allowing font size to change whilst retaining the important boundary - to avoid on-screen text), in practice you can't in browsers.
4/ There is not a good OSS reference implementation.

I'm not saying we should enforce implementation of regions in every browser; but assessing the simpler methods of vertical and horizontal positioning available in the spec, highlighting which ones should be used, when/why, and ensuring that the position and style can match those commonly currently used in quality broadcasters is highly important.

So...  fully support focus on this for interop 2025.  We have a 'fair' spec, and if we had a good target to be achieved (e.g. regions is probably going too far), then the rendering can be put right with not too much effort.

**Comment by wpt-interop** [comment]
Thank you for proposing WebVTT for inclusion in Interop 2025.

We are pleased to let you know that this proposal was accepted as part of the WebVTT investigation area. You can follow the progress of this investigation area on the [Interop 2025 dashboard](https://wpt.fyi/interop-2025).

For an overview of our process, see [proposal selection](https://github.com/web-platform-tests/interop/blob/main/2025/selection-process.md). Thank you for contributing to Interop 2025!

_Posted on behalf of the Interop team._

**Comment by foolip** [comment]
Changing the labels to reflect that this was accepted as an investigation area.
